### PR TITLE
docs: Adding 'Troubleshooting' to the 'Migrating to devfile v2' docs

### PR DESCRIPTION
**Pull request by ibuziuk**
Signed-off-by: Ilya Buziuk <ibuziuk@redhat.com>
Related issue - https://github.com/eclipse/che/issues/21005

image

image

### CODE REVIEW

**Review by l0rd — COMMENTED** [review]
The content is good but we are talking about DevWorkspaces, editors, conversion and an `odo` user may get lost. I think we should set the context: that's about v1 Devfiles written for Eclispe Che or CodeReady Workspaces that got translated into v2 Devfiles (manually[add the link to the guide] or automatically[add the link to the npm lib]). The error that are troubleshooted here are Eclipse Che or OpenShift Dev Spaces (DevWorkspace) workpace startup errors (not odo, OpenShift Console). In the future, we may add a troubleshooting `odo` errors (when testing a migrated devfile) section here.

**Comment by ibuziuk** [comment]
@l0rd thanks for review, I have set a clear context for the troubleshooting section

image

Other tools could contribute to this doc by adding another tool specific section e.g. `odo` errors after the devfile v2 migration

**Review by l0rd — APPROVED** [review]
LGTM

**Comment by ibuziuk** [comment]
@rkratky could you please review
my understanding is that once merged the docs team will cherry-pick it to relevant branches.

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[id="proc_trobleshooting_{context}"]
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:1]:
```suggestion
[id="troubleshooting_{context}"]
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+= Trobleshooting
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:2]:
```suggestion
= Troubleshooting
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+This section describes common problems during the devfile migration and potential solutions.
+
+.{che-prod} workspace startup failures after the migration to devfile v2
+
+- If a workspace fails to start after the conversion try referencing the link:https://quay.io/repository/devfile/universal-developer-image[Universal Developer Image] as the only component in the devfile:
+
+====
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:10]:
```suggestion
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+[source,yaml]
+----
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      memoryLimit: 3Gi
+----
+====
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:19]:
```suggestion
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
+[id="proc_trobleshooting_{context}"]
+= Trobleshooting
+
+This section describes common problems during the devfile migration and potential solutions.
+
+.{che-prod} workspace startup failures after the migration to devfile v2
+
+- If a workspace fails to start after the conversion try referencing the link:https://quay.io/repository/devfile/universal-developer-image[Universal Developer Image] as the only component in the devfile:
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:8]:
```suggestion
Workspace fails to start after the conversion:: Try referencing the link:https://quay.io/repository/devfile/universal-developer-image[Universal Developer Image] as the only component in the devfile:
+
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:9]:
```suggestion
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+[source,yaml]
+----
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      memoryLimit: 3Gi
+----
+====
+
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:20]:
```suggestion
+
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+NOTE: link:https://quay.io/repository/devfile/universal-developer-image[Universal Developer Image] provides runtimes for various languages (including Java, Node.js, Python, PHP, Golang) and tools (including `curl`, `jq`, `git`) for development.
+
+- If the conversion fails with the `Error processing devfile: failed to merge DevWorkspace volumes: duplicate volume found in devfile: volume_name` this means that there are multiple volumes defined in the original devfile with the same name. Try to remove the duplicate volumes and proceed with the conversion.
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:23]:
```suggestion
.Conversion failures

Conversion fails with `Error processing devfile: failed to merge DevWorkspace volumes: duplicate volume found in devfile: volume_name`:: This means there are multiple volumes defined in the original devfile with the same name. Remove the duplicate volumes and proceed with the conversion.
```

File: docs/modules/user-guide/partials/proc_troubleshooting.adoc [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+- If the conversion fails with the `Error provisioning storage: Could not rewrite container volume mounts: volume component 'volume_name' is defined multiple times` this means that the volume defined in the original devfile conflicts with a `volumeMount` propagated by an editor or a plug-in. Try to rename the volume in the defvile to be the same as the name of the `volumeMount` it conflicts with, and proceed with the conversion.
```
Comment (rkratky) [inline docs/modules/user-guide/partials/proc_troubleshooting.adoc:25]:
```suggestion
Conversion fails with `Error provisioning storage: Could not rewrite container volume mounts: volume component 'volume_name' is defined multiple times`:: This means the volume defined in the original devfile conflicts with a `volumeMount` propagated by an editor or a plug-in. Rename the volume in the defvile to be the same as the name of the `volumeMount` it conflicts with, and proceed with the conversion.
```

**Review by rkratky — CHANGES_REQUESTED** [review]
@ibuziuk, thanks. Some suggestions added.

**Comment by ibuziuk** [comment]
@rkratky thanks, the suggestions have been applied

**Review by rkratky — APPROVED** [review]

**Pull request merged**